chalice/deploy/deployer.py

```diff
@@ -280,7 +308,48 @@ def deploy(self, config, existing_resources, stage_name):
                 config, function_name, stage_name)
             deployed_values['api_handler_name'] = function_name
         deployed_values['api_handler_arn'] = function_arn
-        return deployed_values
+
+    def _deploy_auth_handlers(self, config, existing_resources, stage_name,
+                              deployed_values):
+        # type: (Config, OPT_RESOURCES, str, Dict[str, Any]) -> None
+        # The method makes the assumption that _deploy_api_handler
+        # has already been called.  As a result, it reused portions of that
+        # functions configuration:
+        auth_handlers = config.chalice_app.builtin_auth_handlers
+        if not auth_handlers:
+            return
+        for auth_config in auth_handlers:
+            self._deploy_auth_handler(
+                config, auth_config, stage_name, deployed_values)
+
+    def _deploy_auth_handler(self, config, auth_config,
+                             stage_name, deployed_values):
+        # type: (Config, app.BuiltinAuthConfig, str, Dict[str, Any]) -> None
+        api_handler_name = deployed_values['api_handler_name']
+        role_arn = self._get_or_create_lambda_role_arn(
```

Comment:
Yeah.  That would be an (awesome) enhancement.

I'll add support for configuring the role via the `config.json` similar to what we have now, but the autopolicy stuff will just be for the whole app.py file for now.